packages/cli/test/utils.js

```diff
@@ -116,4 +117,116 @@ describe('Utils', () => {
+      const expected = [
+        path.join('tmp', 'app', 'models', 'foo.model.js'),
+        path.join('tmp', 'app', 'models', 'baz.model.js'),
+      ];
+
+      const reader = () => {
+        // Add the expected values to some extras.
+        const result = [
+          path.join('tmp', 'app', 'models', 'bar.js'),
+          path.join('tmp', 'app', 'models', 'README.js'),
+        ].concat(expected);
+        return Promise.resolve(result);
+      };
+
+      return utils
+        .findArtifactPaths(path.join('fake', 'path'), 'model', reader)
+        .then(results => {
+          expect(results).to.eql(expected);
+        });
+    });
+  });
+  describe('getArtifactList', () => {
+    const expectedModels = ['Foo', 'Bar'];
+    const expectedRepos = ['FooRepository', 'BarRepository'];
+
+    getArtifactListTest(
+      'finds JS models',
+      'model',
+      'models',
+      false,
+      [
+        path.join('tmp', 'app', 'foo.model.js'),
+        path.join('tmp', 'app', 'bar.model.js'),
+        path.join('tmp', 'app', 'README.md'),
+      ],
+      expectedModels
+    );
+    getArtifactListTest(
+      'finds TS models',
+      'model',
+      'models',
+      false,
+      [
+        path.join('tmp', 'app', 'foo.model.ts'),
+        path.join('tmp', 'app', 'bar.model.ts'),
+        path.join('tmp', 'app', 'README.md'),
+      ],
+      expectedModels
+    );
+
+    getArtifactListTest(
+      'finds JS repositories',
+      'repository',
+      'repositories',
+      true,
+      [
+        path.join('tmp', 'app', 'foo.repository.js'),
+        path.join('tmp', 'app', 'bar.repository.js'),
+        path.join('tmp', 'app', 'foo.model.js'),
+      ],
+      expectedRepos
+    );
+
+    getArtifactListTest(
+      'finds TS repositories',
+      'repository',
+      'repositories',
+      true,
+      [
+        path.join('tmp', 'app', 'foo.repository.ts'),
+        path.join('tmp', 'app', 'bar.repository.ts'),
+        path.join('tmp', 'app', 'foo.model.ts'),
+      ],
+      expectedRepos
+    );
+
+    /**
+     * Testing function for evaluating the lists returned from
+     * the getArtifactList function.
+     *
+     * @param {string} name The test name
+     * @param {string} artifactType The artifact type under test
+     * @param {string} folder The name of the folder (usually the plural of the
+     * artifactType)
+     * @param {boolean} suffix Whether or not to expect the artifactType as the suffix
+     * (ex. the "foo" repository is FooRepository)
+     * @param {string[]} files An array of fake filepaths to test with
+     * @param {string[]} expected The expected string array of names
+     */
+    function getArtifactListTest(
+      name,
+      artifactType,
+      folder,
+      suffix,
+      files,
+      expected
+    ) {
+      const reader = () => {
+        return Promise.resolve(files);
+      };
+      const artifactPath = path.join('tmp', 'app', folder);
+      it(name, () => {
```

Comment:
My experience with tests generated like this is mixed. Because `it` is called from a helper function, the (async) stack trace reported by a failed test does not show the place where the test suite was defined (where are the test parameters specified).

Can we refactor this helper to remove the call to `it`?

I am envisioning usage like this:

```js
describe('getArtifactList', () => {
  // ...

  it('finds JS repositories', () => {
    return verifyArtifactList(artifactType, folder, suffix, files, expected);
  });

  // ...
});
```

I think it would be even better if we could move `expect(results).to.eql(expected)` out of this helper function too. That way the helper function is focused on executing `getArtifactList` in the given scenario, and everything else is up to the individual test to handle.